src/commonMain/kotlin/spaceEngineers/controller/NextLayerControllerWrapper.kt

```diff
@@ -0,0 +1,65 @@
+package spaceEngineers.controller
+
+import spaceEngineers.commands.*
+import spaceEngineers.model.SeObservation
+import spaceEngineers.model.allBlocks
+
+class NextLayerControllerWrapper(
+    private val controller: CharacterController,
+    val context: SpaceEngineersTestContext = SpaceEngineersTestContext()
+) : CharacterController {
+
+    fun observeForNewBlocks(): SeObservation {
+        return observe(ObservationMode.NEW_BLOCKS).apply {
+            context.updateNewBlocks(allBlocks)
+        }
+    }
+
+    fun startUsingTool(): SeObservation {
+        return interact(InteractionArgs(InteractionType.BEGIN_USE))
+    }
+
+    fun endUsingTool(): SeObservation {
+        return interact(InteractionArgs(InteractionType.END_USE))
+    }
+
+
+    fun equipAndPlace(toolbarLocation: ToolbarLocation): SeObservation {
+        equip(toolbarLocation)
+        return controller.interact(InteractionArgs(InteractionType.PLACE))
+    }
+
+    fun equip(toolbarLocation: ToolbarLocation): SeObservation {
+        return controller.interact(
+            InteractionArgs(
+                InteractionType.EQUIP,
+                slot = toolbarLocation.slot,
+                page = toolbarLocation.page
+            )
+        )
+    }
+
+    fun observerForBlocks(): SeObservation {
+        return observe(ObservationMode.BLOCKS)
+    }
+
+    fun updateObservation(observation: SeObservation) {
```

Comment:
No sure what this does (before looking deep into the code), a candidate for a method commend.